applications/tari_dan_wallet_daemon/src/cli.rs

```diff
@@ -22,11 +22,24 @@
 
 use std::{net::SocketAddr, path::PathBuf};
 
-use clap::Parser;
+use clap::{Args, Parser};
 use minotari_app_utilities::common_cli_args::CommonCliArgs;
 use tari_common::configuration::{ConfigOverrideProvider, Network};
 use url::Url;
 
+#[derive(Args, Debug)]
+pub struct WalletRestoreArgs {
+    #[clap(long)]
+    /// Seed words of a wallet to be restored.
+    /// If set, wallet daemon tries to restore your wallet based on these seed words.
+    pub seed_words: Option<String>,
+
+    #[clap(long)]
+    /// Secret key to be used to encrypt wallet seed key.
+    /// If set this will be used instead an auto generated one.
+    pub wallet_secret: Option<String>,
```

Comment:
This should be using a SafePassword type to prevent leaking the passing in RAM.

Do we need this feature though? Why not always generate the password?

Reply:
just removed